timesource/timesource.go

```diff
@@ -54,13 +56,10 @@ func (e multiRPCError) Error() string {
 	return b.String()
 }
 
-func computeOffset(timeQuery ntpQuery, server string, attempts, allowedFailures int) (time.Duration, error) {
-	if attempts == 0 {
-		return 0, nil
-	}
-	responses := make(chan queryResponse, attempts)
-	for i := 0; i < attempts; i++ {
-		go func() {
+func computeOffset(timeQuery ntpQuery, servers []string, allowedFailures int) (time.Duration, error) {
+	responses := make(chan queryResponse, len(servers))
```

Comment:
Not sure we need some defensive programming here but with `len(servers) == 0`, this function will deadlock on `for response := range responses {`.

Reply:
Good spot @adambabik. It's not a big deal for now, as servers are essentially constants, but you're right it can easily turn out to be a big deal after some refactoring. Adding condition for empty servers slice.